plugins/org.jboss.tools.openshift.egit.core/src/org/jboss/tools/openshift/egit/core/EGitUtils.java

```diff
@@ -887,15 +937,29 @@ public static String getCurrentBranch(Repository repository)
 		return branch;
 	}
 
+	/**
+	 * Returns the current branch for the given project.
+	 * 
+	 * @param project
+	 * @return
+	 * @throws CoreException
+	 * 
+	 * @see #getRepository(IProject)
+	 */
 	public static String getCurrentBranch(IProject project)
 			throws CoreException {
 		if (project == null) {
 			return null;
 		}
 		Repository repo = EGitUtils.getRepository(project);
```

Comment:
I think it's better to use try-with-resources here, it will also fix Sonar complaining about it. I mean it should be smth like:
````java
try (Repository repo = EGitUtils.getRepository(project)) {
    if (repo == null) {
        return null;
    }
    return EGitUtils.getCurrentBranch(repo);
}